spec/syntax/function_spec.rb

```diff
@@ -17,4 +17,58 @@
         __ensure_defimpl__(protocol, for, env)
     EOF
   end
+
+  it 'detects higher order function calls' do
+    expect(<<~'EOF').to include_elixir_syntax('elixirFunctionCall', 'func')
+      func.()
+    EOF
+  end
+
+  it 'detects function calls with parenthesis' do
+    expect(<<~'EOF').to include_elixir_syntax('elixirFunctionCall', 'func')
+      func()
+    EOF
+  end
+
+  it 'detects function calls with bangs' do
+    expect(<<~'EOF').to include_elixir_syntax('elixirFunctionCall', 'func!')
+      func!()
+    EOF
+  end
+
+  it 'detects function calls with question marks' do
+    expect(<<~'EOF').to include_elixir_syntax('elixirFunctionCall', 'func?')
+      func?()
+    EOF
+  end
+
+  it 'detects function calls appended by module with parenthesis' do
+    expect(<<~'EOF').to include_elixir_syntax('elixirFunctionCall', 'func')
+      Mod.func()
+    EOF
+  end
+
+  it 'detects function calls appended by module without parenthesis' do
+    expect(<<~'EOF').to include_elixir_syntax('elixirFunctionCall', 'func')
+      Mod.func
+    EOF
+  end
+
+  it 'does not highlight terms without a parentheresis' do
+    expect(<<~'EOF').not_to include_elixir_syntax('elixirFunctionCall', 'func')
+      func
```

Comment:
This seems odd and I don't know what the solution here is. This causes only some function calls to be highlighted by default:

<img width="198" alt="image" src="https://user-images.githubusercontent.com/6093002/64039836-9840ec00-cb29-11e9-9b6b-876e58b59c34.png">

Both `hd` instances come back as `elixirId` rather than `elixirFunctionCall`

Reply:
I have to confess I completely forgot we could use atoms as module names. Just pushed a commit to fix that.

Also, while at it I realised `|> func` should also be highlighted since if we are piping it is not ambiguous that we are calling a function.